src/classes/Hacker.js

```diff
@@ -11,26 +13,72 @@ export default class Hacker {
     // console.log("Hacker")
   }
 
+  /**
+   * This function will figure out what card the AI should play.
+   * @param event
+   * @returns {*} The card to play, the stack to play, the opponent to attack, and the move type.
+   */
   turnLogic(event) {
+    //This will be executed in either OpponentStacks or PlayerInfoPanel
     let cardToPlay = undefined;
+    //This will be executed in AiMove.js
     let stackToPlay = undefined;
+    //This will be executed in AiMove.js
     let opponentToAttack = undefined;
+    //This is used in mutations under AiTakeTurn
     let moveType = undefined;
 
     let hand = this.move.organizeHand(event);
-
     this.boolSide = store.getters.getCoinMsg;
-    // console.log("The coinMsg: " + store.getters.getCoinMsg);
-    // console.log("AI is choosing: " + this.boolSide);
-
     let canGroup = this.move.findGroup(event.stack, hand.bestGCard);
 
     if(hand.hackCard !== undefined && this.move.getHackOpponent(event) !== undefined && event.stack.find(stack => stack.boolSide === this.boolSide)) {
       cardToPlay = hand.hackCard;
       opponentToAttack = this.move.getHackOpponent(event);
       moveType = 'hack';
+      return {cardToPlay, stackToPlay, opponentToAttack, moveType};
+
+    }
+    else if(hand.powerOutageCard !== undefined){
+      opponentToAttack = this.move.getOpponentToAttack(event);
+      cardToPlay = hand.powerOutageCard;
+      moveType = 'attack'
+    }
+
+    else if(hand.generatorCard !== undefined){
+      cardToPlay = hand.generatorCard;
+      moveType = 'protection'
+    }
+
+    else if(hand.antiVirusCard !== undefined){
+      cardToPlay = hand.antiVirusCard;
+      moveType = 'protection';
+    }
+    else if(hand.firewallCard !== undefined){
+      cardToPlay = hand.firewallCard;
+      moveType = 'protection'
+    }
 
-    } else if(hand.bestVCard !== undefined && this.move.stackToAddVariable(event) !== undefined && event.stack.find(stack => stack.boolSide === this.boolSide)) {
+    else if(hand.batteryBackupCard !== undefined && store.getters.getCurrentPlayer.hasPowerOutage){
+      console.log("Current player: " + JSON.stringify(store.getters.getCurrentPlayer));
```

Comment:
Debug code

Perhaps do a quick search over the files for "console.log" to catch the rest of these. Something like ``grep -r "console.log" src``